cfgov/jinja2/v1/_includes/organisms/filterable-list-controls.html

```diff
@@ -353,6 +348,184 @@
+                method="get"
+                action=".">
+            {% elif 'pdf-generator' in controls.form_type %}
+                method="post"
+                action="pdf/">
+            <input type="hidden" name="csrfmiddlewaretoken" value="{{ csrf_token }}">
+            {% endif %}
+            <input type="hidden" name="form-id" id="form-id" value="{{ index }}">
+
+            <div class="form-l_col
+                        form-l_col-1">
+                <div class="form-group">
+                    <div class="form-l">
+                        {{ _render_filter_fields(controls, form, index) }}
+                    </div>
+                </div>
+            </div>
+            <div class="form-actions form-actions__left-on-med">
+                <input class="btn form-actions_item"
+                       type="submit"
+                       value="Apply Filters">
+                <input class="btn btn__warning btn__link form-actions_item js-form_clear"
+                       type="button"
+                       value="Clear filters"
+                       data-js-hook="form-clear">
+            </div>
+        </form>
+    {% endmacro %}
+
+    {% from 'molecules/notification.html' import render as notification with context %}
+    {% from 'organisms/post-preview.html' import render as post_preview with context %}
+
+    {% macro render(controls, form, index) %}
+        <div class="o-filterable-list-controls">
+            {% set form_markup = _filters_form(controls, form, index) %}
+            {% call() expandable(controls, true) %}
+                {{ form_markup | safe }}
+            {% endcall %}
+            {% for field in form %}
+                {% if field.errors %}
+                    {% for error in field.errors %}
+                        {{ notification('error', true, error) }}
+                    {% endfor %}
+                {% endif %}
+            {% endfor %}
+            {% if form.non_field_errors() %}
+                {% for error in form.non_field_errors() %}
+                        {{ notification('error', true, error) }}
+                {% endfor %}
+            {% endif %}
+            {% if 'filterable-list' in controls.form_type and posts is defined %}
+                {% if posts.paginator.count == 0 %}
+                    {{ notification('warning', true, 
+                        'Sorry, there were no results based on your filter selections.',
+                        'Please reset the filter or change your selections and try again.') }}
+                {% else %}
+                    {{ notification('success', true, posts.paginator.count ~ ' filtered results') }}
+                {% endif %}
+            {% endif %}
+            {% for post in posts %}
+                {{ post_preview(post, controls.categories.page_type, index) }}
+            {% endfor %}
+            {% include 'molecules/pagination.html' %}
+        </div>
+    {% endmacro %}
+
+{% endif %}
```

Comment:
This is a very big big boy. Maybe we can simplify the file some and break it up into reusable pieces.